app/controllers/display_ad_events_controller.rb

```diff
@@ -25,6 +27,6 @@ def update_display_ads_data
   end
 
   def display_ad_event_params
-    params.require(:display_ad_event).permit(%i[context_type category display_ad_id])
+    params[:display_ad_event].slice(:context_type, :category, :display_ad_id)
```

Comment:
I think we can keep `.permit`, you can just remove `require`: https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-permit

Reply:
Not without an additional `include`, to my knowledge.